autonomy/analyse/abci/app_spec.py

```diff
@@ -430,19 +429,22 @@ def check_one(informat: str, infile: str, classfqn: str) -> bool:
 
     @classmethod
     def check_all(cls, packages_dir: Path) -> None:
-        """Check all the available definitions."""
+        """
+        Check all the available definitions.
+
+        This function finds all files whose name is 'fsm_specification.yaml' in an AEA skill package, and checks them
+        one by one using the 'check_one' function.
+
+        :param packages_dir: the package directory where to find the spec files.
+        """
 
         did_not_match = []
         fsm_specifications = sorted(
             [
-                *packages_dir.glob(f"**/{DEFAULT_FSM_SPEC_YAML}"),
+                *packages_dir.glob(f"*/skills/*/{DEFAULT_FSM_SPEC_YAML}"),
             ]
         )
-        # TODO: fix this implementation:
-        # - above file names should be constant and explained in helper text
-        # - investigate: either a) we need to do this in the order of the inverse dependency tree, so that
-        # dependencies are already loaded. Otherwise dependencies might not be present. Or b)
-        # we need to ensure relative modules are loaded too (https://docs.python.org/3/library/importlib.html#importlib.import_module)?
```

Comment:
(a) does not hold because we are assuming (1) the package directory is named `packages` and (2) we add its parent directory to the `sys.path` at position `0`, hence giving the highest priority to the modules in `packages` when importing modules.

(b) should hold as a consequence of the above, since we rely on the standard Python import machinery.